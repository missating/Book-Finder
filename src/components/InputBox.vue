<template>
  <v-container>
    <v-form ref="form" v-model="valid" lazy-validation>
      <v-container align-center justify-center>
        <v-layout>
          <v-flex xs12>
            <v-text-field
              label="Search For Books by: Title, Author etc"
              id="input-box"
              v-model="searchQuery"
              :rules="searchQueryRules"
              required
            ></v-text-field>
          </v-flex>
          <v-flex xs12 md4>
            <v-btn :disabled="!valid" raised @click="validate">
              <span class>Search</span>
            </v-btn>
          </v-flex>
        </v-layout>
      </v-container>
    </v-form>
    <v-container v-if="isFetching">
      <div class="text-xs-center">
        <v-progress-circular indeterminate :size="150" :width="8" color="black"></v-progress-circular>
      </div>
    </v-container>
    <v-container v-if="showSearchQuery && !isFetching">
      <p class="text">Showing results for "{{ searchQuery }}"</p>
    </v-container>
  </v-container>
</template>

<script>
export default {
  name: 'InputBox',
  components: {},
  data() {
    return {
      valid: true,
      searchQuery: '',
      searchQueryRules: [v => !!v || 'Title or Author of a book is required'],
      showSearchQuery: false
    };
  },

  methods: {
    validate() {
      if (this.$refs.form.validate()) {
        this.$store.dispatch('getBooks', this.searchQuery);
        this.showSearchQuery = true;
      }
    },
  },
  computed: {
    isFetching() {
      return this.$store.state.isFetching;
    }
  }
};
</script>

<style>
</style>
