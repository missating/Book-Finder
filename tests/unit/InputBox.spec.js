import Vuex from 'vuex';
import Vue from 'vue';
import { mount, createLocalVue } from '@vue/test-utils';
import InputBox from '@/components/InputBox';
import Vuetify from 'vuetify';

Vue.use(Vuetify);
const localVue = createLocalVue();
localVue.use(Vuex);

describe('InputBox.vue', () => {
  let store;
  let actions;

  beforeEach(() => {
    actions = {
      validate: jest.fn(),
    };

    store = new Vuex.Store({
      state: {
        books: [],
        searchQuery: '',
        isFetching: false,
        apiBaseUrl: 'https://www.googleapis.com/books/v1/volumes',
      },
      actions,
    });
  });

  it('should render correctly', () => {
    const wrapper = mount(InputBox, {
      store,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should update the component state when input is changed', () => {
    const wrapper = mount(InputBox, {
      store,
      localVue,
    });

    wrapper.find('#input-box').setValue('alice');
    wrapper.find('form').trigger('change');
    expect(wrapper.vm.searchQuery).toBe('alice');
  });
});
