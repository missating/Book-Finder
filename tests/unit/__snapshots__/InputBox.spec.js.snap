// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox.vue should render correctly 1`] = `
<div
  class="container"
>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container align-center justify-center"
    >
      <div
        class="layout"
      >
        <div
          class="flex xs12"
        >
          <div
            class="v-input v-text-field theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label theme--light"
                    for="input-box"
                    style="left: 0px; position: absolute;"
                  >
                    Search For Books by: Title, Author etc
                  </label>
                  <input
                    id="input-box"
                    required="required"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex xs12 md4"
        >
          <button
            class="v-btn theme--light"
            raised=""
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <span>
                Search
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </form>
   
  <!---->
   
  <!---->
</div>
`;
